src/traces/table/plot.js

```diff
@@ -0,0 +1,892 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var c = require('./constants');
+var d3 = require('d3');
+var gup = require('../../lib/gup');
+var Drawing = require('../../components/drawing');
+var extendFlat = require('../../lib/extend').extendFlat;
+var svgUtil = require('../../lib/svg_text_utils');
+var raiseToTop = require('../../lib').raiseToTop;
+var cancelEeaseColumn = require('../../lib').cancelTransition;
+
+module.exports = function plot(gd, calcdata) {
+
+    if(c.clipView) {
+        gd._fullLayout._paper.attr('height', 2000);
+    }
+
+    var table = gd._fullLayout._paper.selectAll('.table')
+        .data(calcdata.map(gup.unwrap), gup.keyFun);
+
+    table.exit().remove();
+
+    table.enter()
+        .append('g')
+        .classed('table', true)
+        .attr('overflow', 'visible')
+        .style('box-sizing', 'content-box')
+        .style('position', 'absolute')
+        .style('left', 0)
+        .style('overflow', 'visible')
+        .style('shape-rendering', 'crispEdges')
+        .style('pointer-events', 'all'); // todo restore 'none'
+
+    table
+        .attr('width', function(d) {return d.width + d.size.l + d.size.r;})
+        .attr('height', function(d) {return d.height + d.size.t + d.size.b;})
+        .attr('transform', function(d) {
+            return 'translate(' + d.translateX + ',' + d.translateY + ')';
+        });
+
+    var tableControlView = table.selectAll('.tableControlView')
+        .data(gup.repeat, gup.keyFun);
+
+    tableControlView.enter()
+        .append('g')
+        .classed('tableControlView', true)
+        .style('box-sizing', 'content-box')
+        .on('mousemove', function() {tableControlView.call(renderScrollbarKit, gd);})
```

Comment:
Have you thought about `touch` handlers?

Have you looked into [`dragElement`](https://github.com/plotly/plotly.js/blob/master/src/components/dragelement/index.js) to DRY this up?

Reply:
Yes, both column drag&drop and scrolling work on mobile, though on the codepen exampe it also drags the document somewhat so it's not immediately obvious (iOS needs page-level specs so that the document not be dragged around, I can add a demo later). But _kinetic_ scrolling doesn't now work on mobile (not sure of its urgency).

I'll look into `dragElement` to see if it'd yield code saving or unification benefit. On an initial look, immersing myself would take significant time as `dragElement` is hundreds of code lines and modifies the DOM so may or may not be a good fit (as I don't know it it's hard to tell).

So maybe, as there are other, functional gaps / regressions to cover, and a number of important PR feedback items, it could also be a separate PR that would also involve `parcoords` which also has dragging.

Reply:
No need to switch to `dragElement` in this PR. But we should really start centrailizing all drag handlers soon.

Reply:
Good plan, totally warranted! I feel we're pushing the perf boundaries with new plot types, e.g. `parcoords` and `table` which must support an essentially unlimited row count (plain content tested with 1m rows and 20 columns just fine; Bird mock is just 7k for saving space) as well as both doing asynchronous rendering. So I think it's fitting that unification happens once we have diverse experience from all use cases, eg. small legend panels and crazy high performance alike (our preexisting things don't use revolvers etc.)

Reply:
Issue https://github.com/plotly/plotly.js/issues/2057 created for `dragElement` related unification